ovh/data_hosting_privatedatabase.go

```diff
@@ -0,0 +1,157 @@
+)
+
+func dataSourceHostingPrivateDatabase() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceHostingPrivateDatabaseRead,
+		Schema: map[string]*schema.Schema{
+			"service_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			// Computed
+			"cpu": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Number of CPU on your private database",
+			},
+			"datacenter": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Datacenter where this private database is located",
+			},
+			"display_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name displayed in customer panel for your private database",
+			},
+			"hostname": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Private database hostname",
+			},
+			"hostname_ftp": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Private database FTP hostname",
+			},
+			"infrastructure": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Infrastructure where service was stored",
+			},
+			"offer": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Type of the private database offer",
+			},
+			"port": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Private database service port",
+			},
+			"port_ftp": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Private database FTP port",
+			},
+			"quota_size": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Space allowed (in MB) on your private database",
+			},
+			"quota_used": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Space used (in MB) on your private database",
+			},
+			"ram": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Amount of ram (in MB) on your private database",
+			},
+			"server": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Private database server name",
+			},
+			"state": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Private database state",
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Private database type",
+			},
+			"version": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Private database available versions",
+			},
+			"version_label": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Private database version label",
+			},
+			"version_number": {
+				Type:        schema.TypeFloat,
+				Computed:    true,
+				Description: "Private database version number",
+			},
+		},
+	}
+}
+
+func dataSourceHostingPrivateDatabaseRead(d *schema.ResourceData, meta interface{}) error {
```

Comment:
Small question:

What about setting the endpoint and dynamically the fields, like other datasources?
Example:

```
func dataSourcexxxxRead(d *schema.ResourceData, meta interface{}) error {
	config := meta.(*Config)
	serviceName := d.Get("service_name").(string)

	endpoint := fmt.Sprintf("/xx/xx/%s/kube/%s/xx", serviceName)
	res := &xxxxResponse{}

	log.Printf("[DEBUG] Will read xx from xx %s and xx: %s", xx, serviceName)
	err := config.OVHClient.Get(endpoint, res)
	if err != nil {
		return fmt.Errorf("calling get %s %w", endpoint, err)
	}
	for k, v := range res.ToMap() {
		if k != "id" {
			d.Set(k, v)
		}
	}
	d.SetId(xxId + "-" + res.ClientID + "-" + res.IssuerUrl)

	log.Printf("[DEBUG] Read xx %+v", res)
	return nil
} 
```